### DISCUSSION TITLE
GoogleTranslator maximum number of characters

### DISCUSSION THREAD

**Discussion by LivingDeadCloud** [nidhaloff/deep-translator#100] [Q&A]
Hi there

I just found this packages and I'm starting to use it. Good job!

My question is, is there a limit if I want to translate text while using the sub.module GoogleTranslator? And if there is, how much is this limit?
I have to translate a HUGE amount of text (like 20 billion characters), and I would like to know if this solution is viable. I already investigated DeepL but it would be too expensive.

Thanks everybody

**Comment by ireneb612** [comment]
@LivingDeadCloud Did you manage to translate that amount of text?

**Reply by LivingDeadCloud** [reply]
Yes I was, with some tweaks, like for example the API can't translate texts longer than 5k characters, so you have to limit it. But it's totally feasible!

**Reply by ireneb612** [reply]
Did you use DeepL ? And thank you! I have less than 5k worlds but I hope I won't need to call it too many times :)

**Reply by LivingDeadCloud** [reply]
If I recall correctly (it was some month ago) I used deep-translator. Tomorrow I will check it to confirm!

**Reply by LivingDeadCloud** [reply]
I can confirm that I used `GoogleTranslator` from the `deep_translator` module:


`
from deep_translator import GoogleTranslator`

**Comment by drdoof2019** [comment]
This is my solution for 5k character limit. 
```
for el in text_list:
        if len(el) < 5000:
            try:
                translated_text = translate.translate(el)
            except:
                pass
            else:
                write_to_file(el,translated_text)
        else:
            el_list = el.split('.')
            sep_text = ""
            part_translated_text = ""
            j = 0
            while True:
                while len(sep_text)<4000:
                    try:
                        sep_text = sep_text + el_list[j]
                        j += 1
                    except:
                        break
                part_translated_text = part_translated_text + translate.translate(sep_text)
                if sep_text == "":
                    write_to_file(el,part_translated_text)
                    break
                sep_text = ""
```
Note: If you copy paste the code, it won't work because you dont have necessary functions. I did not delete them because reader can understand whats going on. Just delete unnecessary parts before use.
It basically splits text with "." and add them until they reach +4000 characters then translate first part then it executes second part etc.. until there are no more part to translate.

**Comment by nidhaloff** [comment]
Sorry for the late reply. The limit is translator-dependent. If you google for example the limit for google translate then you can find that it is 5k characters, etc...
